build_resolvers/lib/src/resolver.dart

```diff
@@ -151,6 +155,23 @@ class AnalyzerResolver implements ReleasableResolver {
         (await _driver.getSourceKind(assetPath(assetId))) == SourceKind.LIBRARY;
   }
 
+  @override
+  Future<AstNode> astNodeFor(Element element, {bool resolve = false}) async {
+    var session = _driver.currentSession;
+    var library = element.library;
+
+    if (resolve) {
+      return (await session.getResolvedLibraryByElement(library))
+          .getElementDeclaration(element)
```

Comment:
@scheglov is it possible for this call to fail with an `InconsistentAnalysisException`? I could drop support for resolved ast nodes if it might.

Reply:
I would be in favor of starting without support for resolved nodes unless we know someone is asking for it today.

Reply:
I added a test and it seems fine to use these results after the session is invalidated

Reply:
As it is now, `getElementDeclaration` never fails with `InconsistentAnalysisException`. Parsed/ResolvedLibraryResult has all its parsed/resolved units, and its own copy of the element model.

But ideally my dream is to keep a single copy of the element model, and do fine grained patching as some declarations in files change. This will be a breaking change of course.

Reply:
Using this from reflectable, I get null errors at `.node`. The code looks like what I have in reflectable now, except for `?.node` at this point rather than the plain `.node`.

Reply:
Looks like line 168 could have the same issue.

Reply:
I thought I had checked this case and this returned essentially an "empty" element declaration, and not `null`. But I certainly might have missed something here, I will add the `?.`.